src/Framework/Numerical/Spline.h

```diff
@@ -128,6 +142,10 @@ class Spline : public TObject {
   double     fYMax;
   TSpline3 * fInterpolator;
   bool       fYCanBeNegative;
+  TSpline5 * fInterpolator5;
+  ROOT::Math::Interpolator * fGSLInterpolator;
+  string     fInterpolatorType;
+  std::vector<double> fX, fY;
```

Comment:
Indeed, I considered two solutions:
1. get all knots for spline from `TSpline3::GetKnot(i, x, y)' in the `genie::Spline::SetType`
2. save x- and y-values in the vectors

Reply:
A "typical" file in use currently for Fermilab experiments use, has, in order to cover all the use cases 6 neutrino flavors (nu_e, nu_mu, nu_tau + anti-neutrinos) and  ~46 isotopes (~114 for the larger files).   The number of isotopes seems to slowly grow as people incorporate more details into their geometries.   I've also normally generated 250 knots (rather than the 100 you assume) to better cover the range up to 1000 GeV.

A grep on one such file has 29480 actual splines,  so 29480*8*250*2 = 112 MB which out of a grid node 2GB (about 5% extra) isn't huge but isn't negligible either when in the context of say running GENIE under LArSoft which has its own significant overheads and has memory pressures.

Reply:
Fixed